Display single restaurant card with image

karen-olson opened the pull request:
The following changes are included:

- Refactor to move the `RestaurantCard` component into the Components folder
   - This refactor included removing the `/restaurants` route as everything is now on the home page
- Display the restaurant's image
- Add minimal styling to the card
- Display a single card when the page is loaded (instead of a list of all cards)

Tests for the `CardScreen` component have not yet been updated to include a test for displaying a single card rather than a list.

evan-m-jackson committed:
Prettify the App

- Add `components` directory
- Remove default Next.js design and begin creating main page design

seoulection committed:
adds accept/decline buttons

Commit:
Displays restaurant images and adds minimal styling

Commit:
Adds image and displays a single card

Changes include a refactor to bring Card
component into component folder

karen-olson commented:
I'm having some trouble with configuring next to display the images. I'll be out tomorrow, so if anyone wants to take a stab at this, please feel free! Thanks!

I was getting an error message in the browser saying this:
- `Unhandled Runtime Error
Error: Invalid src prop [(https://s3-media0.fl.yelpcdn.com/bphoto/UvE_Gc-vE1RUO4hCeR-7eQ/300s.jpg)](http://localhost:3000/(https://s3-media0.fl.yelpcdn.com/bphoto/UvE_Gc-vE1RUO4hCeR-7eQ/300s.jpg)) on `next/image`, hostname "s3-media0.fl.yelpcdn.com" is not configured under images in your `next.config.js`
See more info: https://nextjs.org/docs/messages/next-image-unconfigured-host`

I added this code block to `next.config.js` to try to solve it:
- `module.exports = {
    images: {
        remotePatterns: [
            {
                protocol: 'https',
                hostname: '**',
            },
        ],
    },
}`

Then, when trying to run in dev mode, the app failed to start due to this error:
- `error Failed to load next.config.js, see more info here https://nextjs.org/docs/messages/next-config-error
file:///Users/kolson/code/tastemate/frontend/next.config.js:4
module.exports = {
^
ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/kolson/code/tastemate/frontend/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///Users/kolson/code/tastemate/frontend/next.config.js:4:1`

I converted next.config.js to a .cjs file, but still got the same error I'd had initially regarding the unconfigured hostname. I also tried configuring next.config.cjs with the exact hostname the error said it couldn't find, but didn't have any luck. 

I suspect that something may not be successfully reading next.config.cjs, either the whole file or only the module.exports I just added. 

Thanks in advance for any help!

seoulection committed:
makes images work

seoulection commented:
@karen-olson I went ahead a pushed a commit 😅 you were on the right path. We just needed to add the configs to the `nextConfig` object instead of using `module.exports` in `next.config.js`

![Screenshot 2023-09-21 at 10 50 30 AM](https://github.com/8l-project-jam/tastemate/assets/6445410/f1caa344-e9ad-49ab-a686-5251597ea557)

seoulection commented on frontend/src/components/RestaurantCard.tsx:
```diff
@@ -40,7 +41,7 @@ export default function RestaurantCard({ onAcceptButtonClick, onDeclineButtonCli
                   <p>{rating}</p>
                   <div className="h-48 w-96">
-                      <img className="object-cover h-48 w-72" src={image_url} />
+                      <Image className="object-cover h-48 w-72" src={image_url} alt="hi" width={50} height={50} />
```
using next's image component here. we'd probably want to modify the width and height, but those (along with `alt`) are required attributes so i just threw in dummy data for now

clairenguyen commented on frontend/__tests__/components/RestaurantCard.test.tsx:
```diff
@@ -18,12 +18,13 @@ describe('RestaurantCard', () => {
     expect(screen.getByText(FAKE_RESTAURANT.phone)).toBeInTheDocument()
-    expect(screen.getByText(FAKE_RESTAURANT.image_url)).toBeInTheDocument()
-    expect(screen.getByText(FAKE_RESTAURANT.website_url)).toBeInTheDocument()
+    const image = document.querySelector("img") as HTMLImageElement
```
i think we should be able to still use screen/react testing library for this instead of document.querySelector --> `screen.getByRole("img")`.

[Example from the docs](https://testing-library.com/docs/queries/byrole/#api)
`
For example <img aria-label="fancy image" src="fancy.jpg" /> will be returned for getByRole('img', { name: 'fancy image' }). 
`

and then we can assert on the src.  In addition, we can also assert on the `alt` text using screen.*ByAltText`

clairenguyen reviewed (commented):
Nice job @karen-olson  and @seoulection!

Took a look! 
- Added feedback how to find an image using RTL
- I noticed some of the spacing changed from 2 to 4. We probably need to make sure we're using the same prettier package when developing and/or make sure to run `npm run prettier:fix` before pushing either manually or through a pre-push/commit hook.  I created at draft ticket for if/when we wanna implement that. 


I can help add the missing test and run prettier before we merge.

clairenguyen committed:
Run prettier on all the files

clairenguyen committed:
Separate out fetching and display logic into CardScreen and CardCarousel
  - Rename CardScreen.test.tsx to CardCarousel.test.tsx
  - Back-fill CardCarousel and RestaurantCard tests
  - Skip fetch-related tests as it'll be addressed in a separate PR

clairenguyen committed:
Update prettierignore

seoulection committed:
adds fetch tests (#46)
